lighthouse-core/test/runner-test.js

```diff
@@ -295,4 +296,96 @@ describe('Runner', () => {
       assert.strictEqual(AuditClass.meta.name, auditExpectedName);
     });
   });
+
+  it('results include artifacts when given artifacts and audits', () => {
+    const url = 'https://example.com';
+    const config = new Config({
+      audits: [
+        'is-on-https'
+      ],
+
+      artifacts: {
+        HTTPS: {
+          value: true
+        }
+      }
+    });
+
+    return Runner.run({}, {url, config}).then(results => {
+      if (!results.artifacts instanceof Object) {
+        assert.ok(false);
+        return;
+      }
+
+      assert.strictEqual(results.artifacts.HTTPS.value, true);
+
+      for (const method in computedArtifacts) {
+        if (!computedArtifacts.hasOwnProperty(method)) {
+          continue;
+        }
+        assert.ok(results.artifacts.hasOwnProperty(method));
+      }
+    });
+  });
+
+  it('results include artifacts when given passes and audits', () => {
+    const url = 'https://example.com';
+    const config = new Config({
+      passes: [{
+        gatherers: ['https']
+      }],
+
+      audits: [
+        'is-on-https'
+      ]
+    });
+
+    return Runner.run(null, {url, config, driverMock}).then(results => {
+      if (!results.artifacts instanceof Object) {
+        assert.ok(false);
+        return;
+      }
+
```

Comment:
I mean in the other two tests you check `assert.strictEqual(results.artifacts.HTTPS.value, true);`. I'm not sure what the value of `results.artifacts.HTTPS.value` will be here, but I believe runner will attempt to run the `HTTPS` gatherer with the mock driver supplied, so it seems like it's worth keeping the check in there since this is checking the artifacts coming through